server/controllers/conversation.js

```diff
@@ -0,0 +1,125 @@
+const Conversation = require("../models/Conversation");
+const asyncHandler = require("express-async-handler");
+
+// @route POST /create-conversation/
+// @desc create a new conversation
+// @access Public
+
+exports.createConversation = asyncHandler(async (req, res, next) => {
+  const { description, receiver } = req.body;
+
+  if (!description || !receiver) {
+    res.status(400);
+    throw new Error("Bad request! Missing description or receiver!");
+  }
+
+  const conversation = await Conversation.create({
+    messages: {
+      sender: req.user.id,
+      receiver,
+      description,
+    },
+    participants: [req.user.id, receiver],
+  });
+
+  res.status(200).json({
+    success: {
+      conversation: conversation,
+    },
+  });
+});
+
+// @route GET /message/:conversationId
+// @desc get all messages from a single conversation
+// @access Public
+
+exports.getAllMessages = asyncHandler(async (req, res, next) => {
+  const { conversationId } = req.params;
+
+  const conversation = await Conversation.findById(conversationId).populate({
+    path: "messages",
+    sort: { timestamps: -1 },
+  });
+
+  if (conversation.length === 0) {
+    res.status(403);
+    throw new Error("No conversation found!");
+  }
```

Comment:
This is not necessary. When there're no messages, the empty array is returned to the client, then it's up to the client to decide what to render to the user.

Reply:
Okay I removed these from all routes.